great_expectations/execution_engine/pandas_execution_engine.py

```diff
@@ -154,6 +155,13 @@ def get_batch_data_and_markers(
             reader_fn = self._get_reader_fn(reader_method, s3_url.key)
             buf = BytesIO(s3_object["Body"].read())
             buf.seek(0)
+
+            available_params = reader_fn.__code__.co_varnames
+            reader_options = {
+                param: reader_options[param]
+                for param in available_params
+                if param in reader_options
+            }
```

Comment:
Could you add a `logger.warning` message notifying the user of which option keys were dropped at this point?